### PR TITLE
Pre-requsite cells for associated companies

### PR DESCRIPTION
**Pull request by Omri-Levy**
### Description
1. Created a dialog cell using a new re-usable dialog component
2. Created a new dumb call to action cell
3. Added indicators that previous instances of call to action are deprecated
4. Approve action for KYB tasks, directors, and KYC tasks now use the new call to action cell

### CODE REVIEW

File: apps/backoffice-v2/src/domains/entities/hooks/mutations/useRevisionCaseAndDocumentsMutation/useRevisionCaseAndDocumentsMutation.tsx [hunk 1]
```diff
@@ -5,17 +5,11 @@ import { fetchWorkflowEventDecision } from '../../../../workflows/fetchers';
...
 import { Action } from '../../../../../common/enums';
 
-export const useRevisionCaseAndDocumentsMutation = ({
-  workflowId,
-  revisionReason,
-}: {
-  workflowId: string;
-  revisionReason: string;
-}) => {
+export const useRevisionCaseAndDocumentsMutation = ({ workflowId }: { workflowId: string }) => {
```
Comment (Omri-Levy) [inline apps/backoffice-v2/src/domains/entities/hooks/mutations/useRevisionCaseAndDocumentsMutation/useRevisionCaseAndDocumentsMutation.tsx:8]: Required so arguments can be passed from higher up in the blocks API.

File: apps/backoffice-v2/src/pages/Entity/components/CallToActionLegacy/CallToActionLegacy.tsx [hunk 2]
```diff
@@ -0,0 +1,314 @@
...
+import { ICallToActionLegacyProps } from './interfaces';
+
+const motionProps: ComponentProps<typeof MotionButton> = {
+  exit: { opacity: 0, transition: { duration: 0.2 } },
+  initial: { y: 10, opacity: 0 },
+  transition: { type: 'spring', bounce: 0.3 },
+  animate: { y: 0, opacity: 1, transition: { duration: 0.2 } },
+};
+
+export const CallToActionLegacy: FunctionComponent<ICallToActionLegacyProps> = ({ value }) => {
```
Comment (Omri-Levy) [inline apps/backoffice-v2/src/pages/Entity/components/CallToActionLegacy/CallToActionLegacy.tsx:27]: We will no longer use this component once dialogs are separated from buttons and the directors logic is separated out. (adding a task)

File: apps/backoffice-v2/src/pages/Entity/components/CallToActionLegacy/hooks/useCallToActionLegacyLogic/useCallToActionLegacyLogic.tsx [hunk 3]
```diff
@@ -145,9 +145,11 @@ export const useCallToActionLogic = (params: UseCallToActionLogicParams) => {
...
 
   const handleDialogClose = useCallback(
-    (open: boolean) => {
+    (isOpen: boolean) => {
       // Calling callback only when dialog is closed.
-      if (open === false) onDialogClose && onDialogClose();
+      if (isOpen || !onDialogClose) return;
+
+      onDialogClose();
     },
```
Comment (Omri-Levy) [inline apps/backoffice-v2/src/pages/Entity/components/CallToActionLegacy/hooks/useCallToActionLegacyLogic/useCallToActionLegacyLogic.tsx:153]: Readability boyscout.

File: apps/backoffice-v2/src/pages/Entity/components/CaseCallToActionLegacy/interfaces.ts [hunk 4]
```diff
@@ -1,16 +1,16 @@
...
     childWorkflowId: string;
     childWorkflowContextSchema: TWorkflowById['childWorkflows'][number]['workflowDefinition']['contextSchema'];
     disabled: boolean;
     decision:
-      | CommonWorkflowStates.REJECTED
-      | CommonWorkflowStates.APPROVED
-      | CommonWorkflowStates.REVISION;
+      | typeof CommonWorkflowStates.REJECTED
+      | typeof CommonWorkflowStates.APPROVED
+      | typeof CommonWorkflowStates.REVISION;
```
Comment (Omri-Levy) [inline apps/backoffice-v2/src/pages/Entity/components/CaseCallToActionLegacy/interfaces.ts:14]: Was previously broken as `CommonWorkflowStates` is a runtime value and not a type.

File: apps/backoffice-v2/src/pages/Entity/components/DialogCell/DialogCell.tsx [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+import { Dialog } from '@/common/components/molecules/Dialog/Dialog';
+
+export interface IDialogCellProps {
+  type: 'dialog';
+  value: ComponentProps<typeof Dialog> & {
+    props?: Omit<ComponentProps<typeof Dialog>, 'onOpenChange' | 'trigger' | 'content' | 'close'>;
+  };
+}
+
+export const DialogCell: FunctionComponent<IDialogCellProps> = ({ value }) => {
```
Comment (Omri-Levy) [inline apps/backoffice-v2/src/pages/Entity/components/DialogCell/DialogCell.tsx:11]: Adapts new `Dialog` component to the blocks API.

File: apps/backoffice-v2/src/pages/Entity/components/KycBlock/hooks/useKycBlock/useKycBlock.tsx [hunk 6]
```diff
@@ -139,7 +145,27 @@ export const useKycBlock = ({
...
+    filterId,
+  });
+  const { data: session } = useAuthenticatedUserQuery();
+  const caseState = useCaseState(session?.user, parentWorkflow);
+  const isDisabled =
+    !caseState.actionButtonsEnabled ||
+    !childWorkflow?.tags?.includes(StateTag.MANUAL_REVIEW) ||
+    noAction ||
+    isLoadingApproveCase ||
+    isLoadingRevisionCase;
```
Comment (Omri-Levy) [inline apps/backoffice-v2/src/pages/Entity/components/KycBlock/hooks/useKycBlock/useKycBlock.tsx:168]: Hoisted logic from `useCaseCallToActionLogic` - will be encapsulated at a later time.

File: apps/backoffice-v2/src/pages/Entity/components/KycBlock/hooks/useKycBlock/useKycBlock.tsx [hunk 7]
```diff
@@ -202,25 +228,23 @@ export const useKycBlock = ({
...
-          childWorkflowContextSchema: childWorkflow?.workflowDefinition?.contextSchema,
-          disabled: isDisabled,
-          approvalStatus: 'approved',
+        type: 'callToAction',
+        value: {
+          text: 'Approve',
+          onClick: onMutateApproveCase,
+          props: {
+            disabled: isDisabled,
+          },
```
Comment (Omri-Levy) [inline apps/backoffice-v2/src/pages/Entity/components/KycBlock/hooks/useKycBlock/useKycBlock.tsx]: `caseCallToAction` is only needed for dialog flows.

File: apps/backoffice-v2/src/pages/Entity/hooks/useTasks/useTasks.tsx [hunk 8]
```diff
@@ -232,6 +235,22 @@ export const useTasks = ({
...
+        taskId,
+        contextUpdateMethod,
+      }: {
+        taskId: string;
+        contextUpdateMethod: 'base' | 'director';
+      }) =>
+      () =>
+        mutateApproveTaskById({ documentId: taskId, contextUpdateMethod }),
+    [mutateApproveTaskById],
+  );
```
Comment (Omri-Levy) [inline apps/backoffice-v2/src/pages/Entity/hooks/useTasks/useTasks.tsx:252]: Hoisted logic from `useCallToActionLogic` - will be encapsulated at a later time.

**Review by chesterkmr — APPROVED** [review]
LGTM🥏

**Pull request merged**